Client/mods/deathmatch/logic/CClientPed.cpp

```diff
@@ -607,8 +607,14 @@ void CClientPed::Teleport(const CVector& vecPosition)
                         SetFrozenWaitingForGroundToLoad(true);
                 }
 
+                // Player has jetpack?
+                bool hasJetpack = HasJetPack();
+
                 // Set the real position
                 m_pPlayerPed->Teleport(vecPosition.fX, vecPosition.fY, vecPosition.fZ);
+
+                // Restore jetpack
+                SetHasJetPack(hasJetpack);
```

Comment:
what about passing m_bHasJetPack directly and removing bool hasJetpack = HasJetPack();
`SetHasJetPack(m_bHasJetPack);`

Reply:
I do not get it I tried with the code while passing  SetHasJetPack(m_bHasJetPack);

```lua
local x, y, z = getElementPosition ( localPlayer )
		
		setElementModel (localPlayer, math.random(255) )
		
		print(isPedWearingJetpack (localPlayer))
``` 
it working normally when checking for the jetpack

Reply:
My mistake - As you can read in the title and in the indicated issue, the problem concerns changing the position with the jetpack